shell/platform/darwin/macos/framework/Headers/FlutterPluginRegistrarMacOS.h

```diff
@@ -52,6 +52,13 @@ FLUTTER_DARWIN_EXPORT
 - (void)addMethodCallDelegate:(nonnull id<FlutterPlugin>)delegate
                       channel:(nonnull FlutterMethodChannel*)channel;
 
+/**
+ * Registers the plugin as a receiver of `NSApplicationDelegate` calls.
+ *
+ * @param delegate The receiving object, such as the plugin's main class.
+ */
+- (void)addApplicationDelegate:(nonnull NSObject<FlutterAppLifecycleDelegate>*)delegate;
```

Comment:
This is a slight deviation from iOS, which uses `FlutterPlugin` as the argument type, but that shouldn't prevent any code sharing.